server/src/main/java/org/opensearch/search/aggregations/metrics/SumAggregator.java

```diff
@@ -118,6 +130,28 @@ public void collect(int doc, long bucket) throws IOException {
         };
     }
 
+    private LeafBucketCollector getStarTreeLeafCollector(LeafReaderContext ctx, LeafBucketCollector sub, CompositeIndexFieldInfo starTree)
+        throws IOException {
+        final BigArrays bigArrays = context.bigArrays();
+        final CompensatedSum kahanSummation = new CompensatedSum(0, 0);
+
+        StarTreeValues starTreeValues = getStarTreeValues(ctx, starTree);
+
+        //
+        String fieldName = ((ValuesSource.Numeric.FieldData) valuesSource).getIndexFieldName();
+
+        return new LeafBucketCollectorBase(sub, starTreeValues) {
+            @Override
+            public void collect(int doc, long bucket) throws IOException {
+                // TODO: Fix the response for collecting star tree sum
+                sums = bigArrays.grow(sums, bucket + 1);
```

Comment:
Can we extract out the default implementation `getDefaultLeafCollector` and reuse the same logic. 

I really like the approach of reusing the existing aggregators.

Reply:
Yes, I'll try and do that, I am inclined to on refactoring & re-using same implementations wherever possible.

Reply:
final SortedNumericDoubleValues values = valuesSource.doubleValues(ctx);

we can check if we are able to get sortedNumericDoubleValues , otherwise we need to convert to double for each doc